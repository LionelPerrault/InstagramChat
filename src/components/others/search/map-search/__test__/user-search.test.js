import React from 'react'
import { create } from 'react-test-renderer'
import UserSearch from '../user-search'

describe('User-Search Component', () => {
  const props = {
    id: 7,
    username: 'ghalib',
    firstname: 'Mirza',
    surname: 'Ghalib',
    mutualFollowersCount: 0,
    clicked() {},
  }

  it('should match snapshot', () => {
    const tree = create(<UserSearch {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with mutual followers also hiding firstname & surname', () => {
    const tree = create(
      <UserSearch {...props} mutualFollowersCount={3} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
