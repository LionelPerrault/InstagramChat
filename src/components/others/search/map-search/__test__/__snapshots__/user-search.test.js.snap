// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User-Search Component should match snapshot 1`] = `
<div
  className="s_d_peo"
  onClick={[Function]}
>
  <Link
    className="s_d_p"
    to="/profile/ghalib"
  >
    <img
      src="/users/7/avatar.jpg"
    />
    <div
      className="s_d_c"
    >
      <span
        className="s_d_username"
      >
        ghalib
      </span>
      <span>
        Mirza Ghalib
      </span>
    </div>
  </Link>
</div>
`;

exports[`User-Search Component should match snapshot with mutual followers also hiding firstname & surname 1`] = `
<div
  className="s_d_peo"
  onClick={[Function]}
>
  <Link
    className="s_d_p"
    to="/profile/ghalib"
  >
    <img
      src="/users/7/avatar.jpg"
    />
    <div
      className="s_d_c"
    >
      <span
        className="s_d_username"
      >
        ghalib
      </span>
      <span>
        3 mutual followers
      </span>
    </div>
  </Link>
</div>
`;
